pkg/registry/ggcr_image.go

```diff
@@ -0,0 +1,109 @@
+package registry
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pkg/errors"
+)
+
+type GGCRImage struct {
```

Comment:
Not sure if `GGCR` is how the community refers to the library. We could call it `RegistryImage` possibly? That or even `RemoteImage` 🤷‍♂